build/Band_Representations.jl

```diff
@@ -0,0 +1,310 @@
+    return round.(character_table\XVector)
+
+end
+
+#StringtoKVec=Dict("Γ"=>KVec("0, 0"), "K"=>KVec("1/3, 1/3") , "M"=>KVec("1/2, 0"))
+
+function return_irvec(sgnum::Int64, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    irvec = Int[]
+    for (klab, kv) in klab2kv
+        isspecial(kv) || continue
+        kv_array   = subduce_onto_littlegroups(klab, kv, ρ, sitegroup)
+        append!(irvec, Int.(kv_array))
+    end
+    return irvec
+end
+
+function return_irvec2d(sgnum::Integer, D::Integer=2)
+    sg=spacegroup(sgnum, D);
+    wps=get_wycks(sgnum, D)
+    A=Vector{Vector{Int}}(undef, 0)
+
+    sitegroups=SiteGroup.(Ref(sg), wps)
+
+    maximal_wyckoffs=Array{WyckPos{2}, 1}(undef, 0)
+
+    for maximal_sitegroup in findmaximal(sitegroups)
+        push!(maximal_wyckoffs, maximal_sitegroup.wp)
+    end
+
+    #for wp in wps
+    for wp in maximal_wyckoffs
+       # print(wp)
+        #isspecial(qvec(wp)) || continue 
+        #isspecial(wp) || continue 
+
+        sitegroup=SiteGroup(sg, wp)
+        ρ_max=size(site_symmetry_irreps(sitegroup))[1]
+        #print(ρ_max)
+        for ρ in 1:ρ_max
+            push!(A, return_irvec(sgnum, ρ, sitegroup))
+        end
+    end
+    return hcat(A...)
+end
+
+
+function get_maximal_lgirreps(sgnum::Integer, D::Integer)
+    maximal_lgirreps=Dict{String, Array{LGIrrep{D}, 1}}()
+    for (klab, lgirrep) in get_lgirreps(sgnum, D)
+        isspecial(kvec(lgirrep[1].lg)) || continue
+            push!(maximal_lgirreps, klab => lgirrep)
+    end
+    return maximal_lgirreps
+end
+
+
+function get_maximal_lgs(sgnum::Integer, D::Integer)
+    maximal_lgs=Dict{String, LittleGroup{D}}()
+    for (klab, lg) in get_littlegroups(sgnum, D)
+        isspecial(kvec(lg)) || continue
+            push!(maximal_lgs, klab => lg)
+    end
+    return maximal_lgs
+end
+
+
+function make_bandrep_set(sgnum::Integer; D::Integer=2)
+    #klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_maximal_lgs(sgnum, D))
+
+    irvec2d=return_irvec2d(sgnum, 2)
+    klabs=Array{String, 1}(undef, 0)
+    kvs=Array{KVec{D}, 1}(undef, 0)
+    for (klab, kv) in klab2kv
+        push!(klabs, klab)
+        push!(kvs, kv)
+       # if isspecial(kv)==true 
+      #  end
+    end
+
+    spinful, allpaths, timeinvar=false, false, true
+    
+    #irlabs=[([[lgg.cdml for lgg in lg] for (klab, lg) in get_lgirreps(sgnum, D)]...)...]
+    #irdims=[([[size(lgg.matrices[1])[1] for lgg in lg] for (klab, lg) in get_lgirreps(sgnum, D)]...)...]
+
+    irlabs=[([[lgg.cdml for lgg in lg] for (klab, lg) in get_maximal_lgirreps(sgnum, D)]...)...]
+    irdims=[([[size(lgg.matrices[1])[1] for lgg in lg] for (klab, lg) in get_maximal_lgirreps(sgnum, D)]...)...]
+
+
+    wps=get_wycks(sgnum, D)
+    sg=spacegroup(sgnum, D)
+
+    wps_special=[]
+    sitesyms=[]
```

Comment:
This generates an untyped empty vector, i.e. a `Vector{Any}`: you probably want to do something like `wps_special = String[]` which generates a typed empty vector, with elements of type `String`.

Reply:
Still applies.